lib/main.dart

```diff
@@ -1,10 +1,144 @@
+        importance: NotificationImportance.High,
+        onlyAlertOnce: true,
+      )
+    ],
+    channelGroups: [
+      NotificationChannelGroup(
+          channelGroupkey: 'therabot_prompt_group',
+          channelGroupName: 'Therabot prompt group')
+    ],
+  );
+
+  await notifications.isNotificationAllowed().then((isAllowed) async {
+    if (!isAllowed) {
+      await AwesomeNotifications().requestPermissionToSendNotifications();
+    }
+  });
+}
+
+Future<void> _createTherabotNotification() async {
+  AwesomeNotifications notifications = AwesomeNotifications();
+  // notifications.dispose();
+
+  await _initializeNotifications(notifications);
+  bool shouldCreate = true;
+  var test = await notifications.listScheduledNotifications();
+  for (var i in test) {
+    if (i.content?.groupKey != null &&
+        i.content?.groupKey == 'therabot_prompt_group') {
+      shouldCreate = false;
+      break;
+    }
+  }
+  if (!shouldCreate) return;
+
+  NotificationCalendar notificationSchedule;
+  String localTimeZone = await notifications.getLocalTimeZoneIdentifier();
+
+  notificationSchedule = NotificationCalendar(
+    timeZone: localTimeZone,
+    // hour: 16,
+    // minute: 59,
+    second: 0,
+    allowWhileIdle: true,
+  );
+
+  await notifications.createNotification(
+    content: NotificationContent(
+        groupKey: 'therabot_prompt_group',
+        channelKey: 'TherabotNotifications',
+        id: 0,
+        title: 'Test notification',
+        body: "Initial notification text!"),
+    actionButtons: [
+      NotificationActionButton(
+        buttonType: ActionButtonType.Default,
+        label: 'Test',
+        key: 'test_button',
+        enabled: true,
+      )
+    ],
+    schedule: notificationSchedule,
+  );
+}
+
+void _createTherabotNotificationListener() async {
+  AwesomeNotifications notifications = AwesomeNotifications();
+  int count = 0;
+  int id = 1;
+
+  receivedNotificationStream ??=
+      notifications.displayedStream.listen((receivedNotification) async {
+    log("${receivedNotification.body}");
+    // FirebaseDbService.addMessageData(receivedNotification.body); //TODO
+    NotificationCalendar notificationSchedule;
+    String localTimeZone = await notifications.getLocalTimeZoneIdentifier();
+
+    notificationSchedule = NotificationCalendar(
+      timeZone: localTimeZone,
+      // hour: 16,
+      // minute: 59,
+      second: 0,
+      allowWhileIdle: true,
+    );
+
+    await notifications.createNotification(
+      content: NotificationContent(
+          groupKey: 'therabot_prompt_group',
+          channelKey: 'TherabotNotifications',
+          id: id,
+          title: 'Test notif',
+          body: "New text + $count"),
+      actionButtons: [
+        NotificationActionButton(
+          buttonType: ActionButtonType.Default,
+          label: 'Test',
+          key: 'test_button',
+          enabled: true,
+        )
+      ],
+      schedule: notificationSchedule,
+    );
```

Comment:
So this sends a test notification at the next minute, am I understanding this right?